tasks/PageIteratorTask.md

```diff
@@ -72,5 +72,23 @@ TypeScript
         }; 
         let pageIterator = new PageIterator(client, response, callback); 
         pageIterator.iterate(); 
+        
+        // Example of passing headers with a PageIterator
+        let response: PageCollection = await client.api("/me/events").headers({ "Prefer": 'outlook.timezone= "utc"' }).get(); 
+        let callback = (data) => { 
+            console.log(data); 
+            return true; 
+        }; 
+        var requestOptions = { headers: { "Prefer":  'outlook.timezone= "utc"' } };
+        let pageIterator = new PageIterator(client, response, callback, requestOptions); 
```

Comment:
it's not clear to me why i can't have something more like this instead of having to make the first call and manage whether i should do iteration.  something we should think about as we look at the future implementations.

let iterator = client.api("/me/events").headers({ "Prefer": 'outlook.timezone= "utc"' }).Iterator(callback)
iterator.iterate()

In this way headers are the same across all requests and you don't have different options between the first call and subsequent ones.

Reply:
This would be helpful and could apply as well to the generated requests in Java and .NET.

For example, in https://github.com/microsoftgraph/msgraph-beta-sdk-dotnet/blob/master/src/Microsoft.Graph/Generated/requests/GraphServiceUsersCollectionRequest.cs

```
var iterator = client.Users.Request().GetIterator(func);
iterator.iterate();
```

Reply:
Created [an issue](https://github.com/microsoftgraph/msgraph-sdk-design/issues/37) for this to be tracked.